compat/win32/fscache.c

```diff
@@ -237,6 +238,24 @@ static struct fsentry *fseentry_create_entry(struct fscache *cache,
 			     &(fse->u.s.st_mtim));
 	filetime_to_timespec((FILETIME *)&(fdata->CreationTime),
 			     &(fse->u.s.st_ctim));
+	if (wsl_compat && fdata->EaSize > 0) {
+		WCHAR wpath[MAX_LONG_PATH];
+		HANDLE hnd;
+		struct stat bst;
+		if (xutftowcs_long_path(wpath, buf) >= 0 &&
```

Comment:
Oh, that's sad. We already had the path in wide characters, but I guess it is not NUL-terminated, eh? That'll make it slightly more tricky to do all of this efficiently. My suggestion would be to add yet another parameter that is passed to the function copying the bits from disk: the length. If it is non-zero, we will have to use a _local_ buffer in that function into which the characters will need to be copied and then NUL-terminated manually.

An obvious alternative would be to add a NUL directly in `fdata->FileName`, but IIRC that would overwrite information we want to keep, so it is not _really_ an option.